src/modal/modal.spec.ts

```diff
@@ -633,6 +727,95 @@ describe('ngb-modal', () => {
 			});
 		});
 
+		describe('update window custom class options', () => {
+			it('should render modals with the correct window custom classes after update', () => {
+				const modalInstance = fixture.componentInstance.open('foo', { windowClass: 'bar' });
+				fixture.detectChanges();
+				expect(fixture.nativeElement).toHaveModal('foo');
+				expect(document.querySelector('ngb-modal-window')).toHaveCssClass('bar');
+
+				fixture.componentInstance.update({ windowClass: 'foo' });
+				fixture.detectChanges();
+				expect(document.querySelector('ngb-modal-window')).toHaveCssClass('foo');
+
+				modalInstance.close();
+				fixture.detectChanges();
+				expect(fixture.nativeElement).not.toHaveModal();
+			});
+		});
+
+		describe('update of backdrop custom class options ', () => {
+			it('should render modals with the correct backdrop custom classes after update', () => {
+				const modalInstance = fixture.componentInstance.open('foo', { backdropClass: 'bar' });
+				fixture.detectChanges();
+				expect(fixture.nativeElement).toHaveModal('foo');
+				expect(document.querySelector('ngb-modal-backdrop')).toHaveCssClass('bar');
+
+				fixture.componentInstance.update({ backdropClass: 'my-fancy-backdrop' });
+				fixture.detectChanges();
+				expect(document.querySelector('ngb-modal-backdrop')).toHaveCssClass('my-fancy-backdrop');
+
+				modalInstance.close();
+				fixture.detectChanges();
+				expect(fixture.nativeElement).not.toHaveModal();
+			});
+		});
+
+		// The usage of private property does not seem to be a clean access way isn't it ?
+		// That's why something like : expect(modalInstance).toHaveOption('ariaDescibedBy').toBe('blabla');
+		// or a public getter like   : expect(modalInstance.getOptions()['ariaDescibedBy']).toBe('blabla');
+		// did seem to be a great idea according to me
+		describe('update of ariaDescibedBy options ', () => {
+			it('should render modals with the correct ariaDescibedBy after update', () => {
+				const modalInstance = fixture.componentInstance.open('foo', { ariaDescribedBy: 'aLabel' });
+				fixture.detectChanges();
+				expect(fixture.nativeElement).toHaveModal('foo');
+				expect(modalInstance._windowCmptRef.instance.ariaDescribedBy).toBe('aLabel');
```

Comment:
as discussed you should check attributes in the resulting DOM here

Reply:
Done